src/Storefront/Controller/StorybookController.php

```diff
@@ -0,0 +1,265 @@
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\OrFilter;
+use Shopware\Core\Framework\Feature;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\PlatformRequest;
+use Shopware\Core\SalesChannelRequest;
+use Shopware\Core\System\SalesChannel\Context\AbstractSalesChannelContextFactory;
+use Shopware\Core\System\SalesChannel\Entity\SalesChannelRepository;
+use Shopware\Core\System\SalesChannel\SalesChannelCollection;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Core\System\SalesChannel\SalesChannelEntity;
+use Shopware\Core\System\SalesChannel\SalesChannelException;
+use Shopware\Storefront\Theme\DatabaseSalesChannelThemeLoader;
+use Shopware\Storefront\Theme\ThemeRuntimeConfigStorage;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\Attribute\Route;
+use Twig\Environment;
+use Twig\Error\RuntimeError;
+use Twig\Error\SyntaxError;
+
+#[Package('framework')]
+class StorybookController extends AbstractController
+{
+    private const PARAMETER_DENY_LIST = [
+        'measureEnabled',
+        'backgrounds',
+        'outline',
+        'viewport',
+    ];
+
+    private const ENTITY_PROPERTY_LIST = [
+        'product',
+        'category',
+        'media',
+    ];
+
+    private const BASE_TEMPLATE = <<<TWIG
+        {% set assets = theme_config('assets.css') %}
+        {% for file in assets %}
+            <link rel="stylesheet"
+                href="{{ asset(file, 'theme') }}">
+        {% endfor %}
+    TWIG;
+
+    /**
+     * @internal
+     *
+     * @param SalesChannelRepository<SalesChannelProductCollection> $productRepository
+     * @param EntityRepository<MediaCollection> $mediaRepository
+     * @param EntityRepository<SalesChannelCollection> $salesChannelRepository
+     */
+    public function __construct(
+        private readonly string $environment,
+        private readonly Environment $twig,
+        private readonly SalesChannelRepository $productRepository,
+        private readonly EntityRepository $mediaRepository,
+        private readonly AbstractSalesChannelContextFactory $contextFactory,
+        private readonly EntityRepository $salesChannelRepository,
+        private readonly DatabaseSalesChannelThemeLoader $themeLoader,
+        private readonly ThemeRuntimeConfigStorage $themeRuntimeConfigStorage,
+    ) {
+    }
+
+    /**
+     * @phpstan-ignore shopware.routeScope
+     */
+    #[Route(
+        path: '/storybook/{component}',
+        name: 'storybook.component',
+        defaults: ['auth_required' => false],
+        methods: ['GET']
+    )]
+    public function storybook(Request $request): Response
```

Comment:
```suggestion
    public function storybook(string $component, Request $request): Response
```

with this, the placeholder will be directly injected. getting the component from the attributes is then not necessary anymore `$component = $request->attributes->get('component');`